pkg/ui/workspaces/cluster-ui/src/api/databases/tableIndexesApi.ts

```diff
@@ -88,7 +88,11 @@ export const useTableIndexStats = ({
           stat?.statistics?.stats?.total_rows_read?.toNumber() ?? 0,
         indexRecs:
           data?.index_recommendations
-            ?.filter(rec => rec?.type != null)
+            ?.filter(
+              rec =>
+                rec?.type != null &&
+                rec.index_id === stat.statistics?.key?.index_id,
```

Comment:
do we not also need `rec?` in the second clause like in `rec?.type`?

Reply:
Technically its not needed because if rec was null, `rec?.type != null` would be false and it would short circuit the expression. That being said, I dont think its possible for `rec` to be null unless the status server was explicitly putting nils into the index stats response